src/psyclone/psyir/transformations/matmul2code_trans.py

```diff
@@ -183,39 +233,34 @@ def validate(self, node):
             # transformation is currently limited to Ranges which
             # specify the full extent of the dimension.
             if not vector.is_full_range(0):
-                raise GenerationError(
+                raise TransformationError(
```

Comment:
In contrast to what I said in the first review, perhaps this would be better as a NotImplementedError?

Reply:
Done.